File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        int mapc[26] = {}, res(0);
        for(char& c : chars) mapc[c - 'a']++;
        for(string& word : words)
        {
            int mapw[26] = {};
            bool flag(true);
            for(char& c : word)
            {
                mapw[c - 'a']++;
                if(mapw[c-'a'] > mapc[c - 'a'])
                {
                    flag = false;
                    break;
                }
            }
            if(flag) res+=word.size();
        }
        return res;
    }
};

File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        mapc,res = Counter(chars),0
        for word in words:
            mapw = Counter(word)
            for c in mapw:
                if mapw[c] > mapc[c]:
                    break
            else:
                res+= len(word)
        return res
                
                
            